effekt/shared/src/main/scala/effekt/RecursiveDescent.scala

```diff
@@ -659,8 +659,17 @@ class RecursiveDescent(positions: Positions, tokens: Seq[Token], source: Source)
   */
   def tryExpr(): Term =
     nonterminal:
-      `try` ~> stmt() ~ someWhile(handler(), `with`) match {
-        case s ~ hs => TryHandle(s, hs)
+      `try` ~> stmt() ~ manyWhile(handler(), `with`) ~ finalizerClause(`suspend`, false) ~ finalizerClause(`resume`, true) ~ finalizerClause(`return`, true) ~ finalizerClause(`finally`, true) match {
```

Comment:
For some reason, this breaks the current positions and error messages have the wrong span:
```
Warning:        |[warning] examples/pos/capture/regions.effekt:10:3: Handling effect Dummy, which seems not to be used by the program.
       |  try { prog() } with Dummy { resume(()) }
       |  ^
       |""".stripMargin
=> Diff (- obtained, + expected)
   try { prog() } with Dummy { resume(()) }
-  ^
+  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```